my-project/src/Service/ImagesCollection.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: virtua
+ * Date: 03.08.2018
+ * Time: 10:09
+ */
+
+namespace App\Service;
+
+use App\Entity\Image;
+use App\Entity\ProductCategory;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+
+class ImagesCollection
+{
+
+    private $imagesDirectory;
+
+    public function __construct($imagesDirectory)
+    {
+        $this->imagesDirectory = $imagesDirectory;
+    }
+    public function addingImagesCollection($ProductCategory, $images){
+
+        $parameterValue = $this->imagesDirectory;
+
+        foreach($images as $image)
+        {
+
+            $file = new Image();
+            $ext = $image->guessExtension();
+            $file->setName($image.".".$ext);
+
+            $ProductCategory->addImage($file);
```

Comment:
1. zmienna małą literą. 
2. W Service może być przygotowana kolekcja obrazów, ale dodanie jej do ProductCategory, powinno już być poza nim - serwisy mają być tak uniwersalne jak to tylko możliwe.
3. Po uwzględniego powyższego (zarówno tu jak i w MainImage) proponowałabym połączenie ImagesCollection i MainImage w jeden service z dwoma funkcjami - jedną do pobrania pojedynczego obrazu, drugą do kolekcji. Kod w  tej pętli jest taki sam jak w addingMainImage